op-txpool/conditional_txs.go

```diff
@@ -0,0 +1,184 @@
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/rpc"
+
+	"github.com/prometheus/client_golang/prometheus"
+
+	"golang.org/x/time/rate"
+)
+
+var (
+	authHeaderKey = "x-optimism-signature"
+
+	// errs
+	rateLimitErr             = &oprpc.JsonError{Message: "rate limited", Code: types.TransactionConditionalRejectedErrCode}
+	endpointDisabledErr      = &oprpc.JsonError{Message: "endpoint disabled", Code: types.TransactionConditionalRejectedErrCode}
+	missingConditionalErr    = &oprpc.JsonError{Message: "missing conditional", Code: types.TransactionConditionalRejectedErrCode}
+	invalidAuthenticationErr = &oprpc.JsonError{Message: "invalid authentication", Code: types.TransactionConditionalRejectedErrCode}
+	entrypointSupportErr     = &oprpc.JsonError{Message: "only 4337 Entrypoint contract support", Code: types.TransactionConditionalRejectedErrCode}
+)
+
+type ConditionalTxService struct {
+	log log.Logger
+	cfg *CLIConfig
+
+	limiter             *rate.Limiter
+	backends            map[string]client.RPC
+	entrypointAddresses map[common.Address]bool
+
+	costSummary prometheus.Summary
+	requests    prometheus.Counter
+	failures    *prometheus.CounterVec
+}
+
+func NewConditionalTxService(ctx context.Context, log log.Logger, m metrics.Factory, cfg *CLIConfig) (*ConditionalTxService, error) {
+	backends := map[string]client.RPC{}
+	for _, addr := range cfg.SendRawTransactionConditionalBackends {
+		rpc, err := client.NewRPC(ctx, log, addr)
+		if err != nil {
+			return nil, fmt.Errorf("failed to dial backend %s: %w", addr, err)
+		}
+
+		rpcMetrics := metrics.MakeRPCClientMetrics(addr, m)
+		backends[addr] = client.NewInstrumentedRPC(rpc, &rpcMetrics)
+	}
+
+	limiter := rate.NewLimiter(types.TransactionConditionalMaxCost, int(cfg.SendRawTransactionConditionalRateLimit))
+	entrypointAddresses := map[common.Address]bool{predeploys.EntryPoint_v060Addr: true, predeploys.EntryPoint_v070Addr: true}
+
+	return &ConditionalTxService{
+		log: log,
+		cfg: cfg,
+
+		limiter:             limiter,
+		backends:            backends,
+		entrypointAddresses: entrypointAddresses,
+
+		costSummary: m.NewSummary(prometheus.SummaryOpts{
+			Namespace: MetricsNameSpace,
+			Name:      "txconditional_cost",
+			Help:      "summary of cost observed by *accepted* conditional txs",
+		}),
+		requests: m.NewCounter(prometheus.CounterOpts{
+			Namespace: MetricsNameSpace,
+			Name:      "txconditional_requests",
+			Help:      "number of conditional transaction requests",
+		}),
+		failures: m.NewCounterVec(prometheus.CounterOpts{
+			Namespace: MetricsNameSpace,
+			Name:      "txconditional_failures",
+			Help:      "number of conditional transaction failures",
+		}, []string{"err"}),
+	}, nil
+}
+
+func (s *ConditionalTxService) SendRawTransactionConditional(ctx context.Context, txBytes hexutil.Bytes, cond *types.TransactionConditional) (common.Hash, error) {
```

Comment:
I thought this service was just supposed to verify the auth header and not do anything else?